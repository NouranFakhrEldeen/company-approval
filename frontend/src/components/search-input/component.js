import React, { useEffect, useState, forwardRef } from 'react';
import { useTranslation } from 'react-i18next';


import { capitalizeFirstWord } from '../../helpers';

const searchInput = forwardRef(({
  handleSearch,
  placeholder,
  timeout,
  name,
  className,
  iconsBoolean,
  icons,

}, ref) => {
  const [searchTerm, setSearchTerm] = useState('');
  const debouncedSearchTerm = useDebounce(searchTerm, timeout || 500);
  useEffect(
    () => {
      handleSearch(debouncedSearchTerm);
    },
    // eslint-disable-next-line
    [debouncedSearchTerm]
  );


  const { t: translate } = useTranslation();

  return (
    <div className='input-field-container'>
      {iconsBoolean && <span className='input-icon-cont'>
        <span className={icons} />
      </span>}
      <input
        ref={ref}
        className={className}
        name={name}
        placeholder={placeholder || `${capitalizeFirstWord(translate('search'))}...`}
        onChange={e => setSearchTerm(e.target.value)}
      />
    </div>
  );
});

export const SearchInput = searchInput;

function useDebounce(value, delay) {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);
    return () => {
      clearTimeout(handler);
    };
    // eslint-disable-next-line
  }, [value]);
  return debouncedValue;
}