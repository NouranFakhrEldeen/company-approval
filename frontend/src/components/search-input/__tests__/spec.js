
import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import { SearchInput } from '../component';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: key => key }),
}));


describe('filter deviations form component', () => {
  it('Should render (smoke test)', () => {

    const { asFragment } = render(
      <Router>
        <SearchInput
          className='padding-5'
          placeholder=''
          t={key=>key}
          handleSearch={() => { }}
          timeout={1000} />
      </Router>,
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});