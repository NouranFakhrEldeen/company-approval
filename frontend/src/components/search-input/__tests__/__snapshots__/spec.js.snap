// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter deviations form component Should render (smoke test) 1`] = `
<DocumentFragment>
  <div
    class="input-field-container"
  >
    <input
      class="padding-5"
      placeholder="Search..."
    />
  </div>
</DocumentFragment>
`;
